File: app/src/main/java/ru/merkulyevsasha/news/data/db/ArticleEntity.java
package ru.merkulyevsasha.news.data.db;


import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Index;
import android.arch.persistence.room.PrimaryKey;

import java.util.Date;

@Entity(tableName = "articles", indices = {@Index("sourceNavId"), @Index("search")})
public class ArticleEntity {

    @PrimaryKey(autoGenerate = true)
    private int id;
    private int sourceNavId;
    private String title;
    private String link;
    private String description;
    private Date pubDate;
    private String category;
    private String search;
    private String pictureUrl;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getSourceNavId() {
        return sourceNavId;
    }

    public void setSourceNavId(int sourceNavId) {
        this.sourceNavId = sourceNavId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getPubDate() {
        return pubDate;
    }

    public void setPubDate(Date pubDate) {
        this.pubDate = pubDate;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getSearch() {
        return search;
    }

    public void setSearch(String search) {
        this.search = search;
    }

    public String getPictureUrl() {
        return pictureUrl;
    }

    public void setPictureUrl(String pictureUrl) {
        this.pictureUrl = pictureUrl;
    }
}

File: database/src/main/java/ru/merkulyevsasha/database/entities/ArticleEntity.kt
package ru.merkulyevsasha.database.entities

import android.arch.persistence.room.Entity
import android.arch.persistence.room.Index
import android.arch.persistence.room.PrimaryKey
import java.util.Date

@Entity(tableName = "articles", indices = [Index("sourceName"), Index("search")])
class ArticleEntity(
    @PrimaryKey
    val articleId: Int = 0,
    val sourceName: String,
    val title: String,
    val link: String,
    val description: String,
    val pubDate: Date,
    val category: String,
    val pictureUrl: String,
    val usersLikeCount: Int,
    val usersDislikeCount: Int,
    val usersCommentCount: Int,
    val isUserLiked: Boolean,
    val isUserDisliked: Boolean,
    val isUserCommented: Boolean,
    val search: String
)